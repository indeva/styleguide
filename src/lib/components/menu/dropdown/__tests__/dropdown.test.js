import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from '..';

describe('Link', () => {
  const defaultProps = {
    active: false,
    links: [
      { title: 'Test' },
      { title: 'Test2', url: '#' },
    ],
  };

  it('when is inactive', () => {
    const component = renderer.create(<Dropdown {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when is active should have active modifier', () => {
    const component = renderer.create(<Dropdown {...defaultProps} active />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
