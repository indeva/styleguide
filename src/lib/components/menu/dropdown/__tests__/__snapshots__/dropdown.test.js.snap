// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link when is active should have active modifier 1`] = `
<ul
  className="menu__dropdown menu__dropdown--active"
>
  <li
    className="menu__item menu__item--child"
  >
    <span
      className="menu__link menu__link--child"
      onClick={null}
      role="presentation"
    >
      Test
    </span>
  </li>
  <li
    className="menu__item menu__item--child"
  >
    <a
      className="menu__link menu__link--child"
      href="#"
      target="_self"
    >
      Test2
    </a>
  </li>
</ul>
`;

exports[`Link when is inactive 1`] = `
<ul
  className="menu__dropdown"
>
  <li
    className="menu__item menu__item--child"
  >
    <span
      className="menu__link menu__link--child"
      onClick={null}
      role="presentation"
    >
      Test
    </span>
  </li>
  <li
    className="menu__item menu__item--child"
  >
    <a
      className="menu__link menu__link--child"
      href="#"
      target="_self"
    >
      Test2
    </a>
  </li>
</ul>
`;
